pkg/options/patchtmpl/patch.go

```diff
@@ -234,28 +252,38 @@ func objectApplier(scheme *PatcherScheme, patches []*parsedPatch) options.ObjHan
 
 		deserializer := scheme.Codecs.UniversalDeserializer()
 
-		kubeObj, err := runtime.Decode(deserializer, objByt)
+		kubeObj, decodeErr := runtime.Decode(deserializer, objByt)
 		_, isUnstructured := kubeObj.(*unstructured.Unstructured)
-		if runtime.IsNotRegisteredError(err) || isUnstructured {
-			// Strategic merge patch can't handle unstructured.Unstructured or
-			// unregistered objects, so return an error. Previously, we deferred
-			// to normal JSON Merge-patch, but this behavior was confusing.
-			return nil, fmt.Errorf("while converting object %q of kind %q and apiVersion %q: type not registered in scheme", obj.GetName(), obj.GetKind(), obj.GetAPIVersion())
-		}
-
-		objSchema, err := strategicpatch.NewPatchMetaFromStruct(kubeObj)
-		if err != nil {
-			return nil, fmt.Errorf("while getting patch meta from object %s: %v", string(objByt), err)
-		}
-
+		strategicWillFail := runtime.IsNotRegisteredError(decodeErr) || isUnstructured
+		objSchema, objSchemaErr := strategicpatch.NewPatchMetaFromStruct(kubeObj)
 		for _, pat := range patches {
 			if !canApplyPatch(pat, obj) {
 				continue
 			}
 
-			newObjJSON, err := strategicpatch.StrategicMergeMapPatchUsingLookupPatchMeta(objJSON, pat.jsonMap, objSchema)
-			if err != nil {
-				return nil, fmt.Errorf("while applying patch\n%sto \n%s: %v", pat.raw, objJSON, err)
+			var newObjJSON map[string]interface{}
+			if pat.patchType == bundle.JSONPatch {
+				if objByt, err := converter.FromObject(objJSON).ToJSON(); err != nil {
```

Comment:
I think it would be clearer if you didn't shadow-variables here. objByt is defined above

Reply:
done